src/patchelf.cc

```diff
@@ -1885,9 +1906,186 @@ void ElfFile<ElfFileParamNames>::clearSymbolVersions(const std::set<std::string>
         }
     }
     changed = true;
-    this->rewriteSections();
+    rewriteSections();
 }
 
+struct symstr {
+    const char *str;
+    size_t len;
+};
+
+// Elf32_Off, Elf32_Word, Elf54_Word = unsigned int
+// Elf64_Off, Elf64_Xword = unsigned long (or long long on 32 bit)
+//typedef variant<Elf32_Off*,Elf32_Word*,Elf64_Off*,Elf64_Word*,Elf64_Xword*> symref;
+//typedef variant<unsigned int*, unsigned long*> symref;
+typedef variant<uint16_t*, uint32_t*, uint64_t*> symref;
+
+struct Comparator {
+    using is_transparent = void;
+    bool operator()(const symstr& a, const symstr& b) const
+    {
+        return (a.str+a.len < b.str+b.len);
+    }
+
+    bool operator()(const char* a, const symstr& b) const
+    {
+        return (a < b.str);
+    }
+
+    bool operator()(const symstr& a, const char* b) const
+    {
+        return (a.str+a.len < b);
+    }
+};
+
+#include <argz.h>
```

Comment:
This might be glibc specific. Should we rely on it?

Reply:
Why not?

Reply:
I suspect there are tests in the CI that compile patchelf against musl libc. But maybe I'm wrong. We'll know when the CI runs.

Reply:
Yes. Musl support is needed in general we not just support Linux but also other operating systems.

Reply:
its already removed with the move to string_view.